mailgun/resource_mailgun_domain.go

```diff
@@ -368,13 +368,20 @@ func ReadDomain(d *schema.ResourceData, meta interface{}) error {
 	mg := meta.(*mailgun.MailgunImpl)
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
 	defer cancel()
-	domainName := d.Get("name").(string)
+	domainName := d.Id()
 	mg = mailgun.NewMailgun(domainName, mg.APIKey())
 
 	domainResponse, err := mg.GetDomain(ctx, domainName)
 	if err != nil {
 		return fmt.Errorf("Error Getting mailgun domain Details for %s: Error: %s", d.Id(), err)
 	}
+	if _, ok := d.GetOk("dkim_key_size"); !ok {
+		d.Set("dkim_key_size", 1024)
```

Comment:
Why are these two defaults needed here again? They are already provided in the resource definition here: https://github.com/fretlink/terraform-provider-mailgun/blob/master/mailgun/resource_mailgun_domain.go#L73

Is it because the API returns a wrong information?

Reply:
It is because the api doesn't give this info when it is imported it doesn't set the default and just read so the default neads to be set if the user does not set them in the configuration when the resource is imported

Reply:
The defaults are correct.

Reply:
from the api documentation:
force_dkim_authority	
true or false

If set to true, the domain will be the DKIM authority for itself even if the root domain is registered on the same mailgun account

If set to false, the domain will have the same DKIM authority as the root domain registered on the same mailgun account

The default is false.

dkim_key_size	
1024 or 2048

Set the length of your domain’s generated DKIM key

The default is 1024